pyramid/session.py

```diff
@@ -626,3 +637,46 @@ def SignedCookieSessionFactory(
         reissue_time=reissue_time,
         set_on_exception=set_on_exception,
     )
+
+class BlowfishPickleSerializer(object):
+    """ Encrypting Webob cookie serializer.
+
+    Uses the pickle protocol to dump Python data to bytes, then encrypts
+    the value symmetrically using the Blowfish algorithm.
+
+    To encrypt sessions, pass an instance of this class as the ``serializer``
+    argument to :class:`BaseCookieSessionFactory`.
+
+    .. note::
+
+       Before using this serializer, install ``pyramid[encrypted_cookies]`` to
+       get the necessary dependency (``pycrypto``).
+    """
+    def __init__(self, secret):
+        if not _HAS_CRYPTO:
+            raise ValueError("No crypto:  install 'pycrypto' first!")
+        self.secret = secret
```

Comment:
This secret should NOT be used directly as the key to any symmetric cipher. You would want to extend it using a key derivation function (such as KDF2). You'll also want a unique salt as @mmerickel mentioned (luckily the KDF2 will happily take that too).

By using the secret directly you are making it easier to attack the crypto. KDF2 with 50,000 iterations would be much better, since this would also lengthen the existing key from what the user provided to the key size required by the cipher. This won't increase the entropy of the key, so if the key was only 12 bits of entropy, after KDF you'd still only have 12 bits of entropy, but the key space to search when brute forcing went from for example 20 bits to 128 bits.